ktor-client/ktor-client-cio/jvm/src/io/ktor/client/engine/cio/Endpoint.kt

```diff
@@ -134,9 +134,10 @@ internal class Endpoint(
         pipeline.pipelineContext.invokeOnCompletion { releaseConnection() }
     }
 
-    private suspend fun connect(): Socket {
+    private suspend fun connect(attributes: Attributes? = null): Socket {
```

Comment:
Need's to be designed. It looks like we shouldn't pass the attributes directly here.
It looks like we can make something like `List<HttpClientEngineExtensions>` highlight the purpose.